src/client/client4.js

```diff
@@ -523,8 +524,16 @@ export default class Client4 {
 
     getProfilesInChannel = async (channelId, page = 0, perPage = PER_PAGE_DEFAULT) => {
         this.trackEvent('api', 'api_profiles_get_in_channel', {channel_id: channelId});
+
+        const serverVersion = Client4.getServerVersion();
+        let queryStringObj;
```

Comment:
I think you want to do `this.getServerVersion()`

Reply:
Hold up, we shouldn't be doing any version checking in the client. Couple things:
1. The client is a 'dumb' web utility and shouldn't do any 'smart' casing like this
2. This will break backwards compatibility the same way changing the default behaviour on the server would. If I upgrade to 4.7 server and start using this client in my integration, all of a sudden the sorting on the results when I use `getProfilesInChannel` is different. That's a breaking change and it's very important we keep those out of the client web utilities

The proper way to do this is to add another argument `sort` and default it to an empty string. Then in your action you do the same thing, add another argument `sort` and default it to whatever makes sense (pretty sure it should still be a blank string though as that's the 99% case and sorting by status is the 1% special case)

See how I added sort to getting emojis for example: 
https://github.com/mattermost/mattermost-redux/pull/367/files#diff-6655fe9b9bb92382c9e8084066aa895eR1702

Notice I didn't add the sort option to the action in that PR because I wanted the default case to sort by name because it made sense in that context

Reply:
P.S. I don't think you'll need to do any backwards compatibility version checking anywhere as even if sort is provided in the query string to an old server, it will ignore it and just do the old behaviour

Reply:
Talked this one over with @jwilander and decided we do need to keep this check here because the server currently explicitly rejects requests with the `sort` attribute. Therefore we can only add this attribute on updated servers.